Remove some `js/src/external-components`

tomalec opened the pull request:
### Changes proposed in this Pull Request:

This is a cleanup task, as the fixes for forked components were released.

- Update `@woocommerce/components` to the latest `6.2.0`  from `5.1.2` (7953722)
- Remove forked `ReportFilters` & co. (53ea575)
	as those fixes were released in `@woocommerce/components`:
	- woocommerce/woocommerce-admin#6890,
	- woocommerce/woocommerce-admin#6062
- Set static options in Programs autocomplete, (dda400e)
	as `Search` fix woocommerce/woocommerce-admin#6061 was released.

### Screenshots:

<!--- Optional --->


### Detailed test instructions:

According to the [woocommerce/components/CHANGELOG.md#620](https://github.com/woocommerce/woocommerce-admin/blob/main/packages/components/CHANGELOG.md#620) the majority of changes were around `Search`, `SummaryNumber` and `<Filter>` components. So report pages are at the biggest risk, to be broken.

1. Go to report pages, or any other you may suspect :|
2. Check that it works as before.


### Changelog Note:

> Removed forked `@woocommerce/components`.

### Additional notes:
- Removing `external-components` makes it easier for us to catch up with other dependencies. For example, the latest `@woocommerce/eslint-plugin` complains a lot about external dependencies and experimental components being used. We have those problems also in our components, but this PR reduces a few.

tomalec commented:
I guess, it got unlocked after we bumped required versions to 
```
WP 5.5
WC 5.2
```

in https://github.com/woocommerce/google-listings-and-ads/pull/803

tomalec commented:
Unfortunately, WC `5.2` uses wc-admin at `2.1.5` https://github.com/woocommerce/woocommerce/blob/5.2.0/composer.json#L24
Which still uses `/components` at `5.1.2` https://github.com/woocommerce/woocommerce-admin/blob/v2.1.5/packages/components/package.json#L3

tomalec commented:
@jconroy @eason9487 Do you know why we can't use our own version of `/components` and we have to share it with wc-admin?

I thought that was the main advantage of React over Web Components, that using different versions of the same component is easy and seamless. Is this a technical or policy limitation?

eason9487 commented:
> Do you know why we can't use our own version of `/components` and we have to share it with wc-admin?

I guess it's a consideration of the overall bundle size of js files. But actually, we're still using our own specific version of `@wordpress/components`, maybe it's not a mandatory restriction?

Also, if we can use our own specific dependency versions, there should be fewer compatibility issues with different versions of WordPress and WooCommerce.

tomalec commented:
I'm rebasing this PR on top of https://github.com/woocommerce/google-listings-and-ads/pull/1110 as there we move to WC5.6+ => `@woocommerce/components` ~8.0.0~ 7.1.0+ which has the changes we need https://github.com/woocommerce/woocommerce-admin/blob/v2.6.5/packages/components/CHANGELOG.md

tomalec committed:
Remove forked `ReportFilters` & co.

as those fixes were released in `@woocommerce/components`:
- https://github.com/woocommerce/woocommerce-admin/issues/6890,
- https://github.com/woocommerce/woocommerce-admin/issues/6062
Remove no longer needed dependencies.

eason9487 reviewed (commented):
Tested with WC 5.7 and 5.9, and there are no errors in either.

🚧 
However, [an inappropriate falsy check](https://github.com/woocommerce/woocommerce-admin/blob/v2.5.0-beta.1/packages/components/src/filter-picker/index.js#L171-L178) was later added to the `<FilterPicker>` component, resulting in the free listings program ID `0` being cleared from the URL query `programs` when filtering single program by free listings program.

https://user-images.githubusercontent.com/17420811/144550609-4fdb6f77-2236-4fb2-aa08-2da74e91a7af.mp4

```
Before: page=wc-admin&reportKey=programs&path=%2Fgoogle%2Freports&filter=single_program&programs=0
After:  page=wc-admin&reportKey=programs&path=%2Fgoogle%2Freports&filter=single_program
```

---

One relevant hidden influence is that the `isSubset` check here will not be `true`, but in terms of results, it will still fall into the `else` logic, making the correct rendering result of single program filter. It can be reproduced by refresh the page with path `/wp-admin/admin.php?page=wc-admin&reportKey=programs&path=%2Fgoogle%2Freports&filter=single_program`.

https://github.com/woocommerce/google-listings-and-ads/blob/49244468e28cf61959a52a21eed56e3e4470d135/js/src/reports/programs/filter-config.js#L74-L82

Another influence is that it still queries the APIs for paid programs and shows incorrect reporting results.

![2021-12-03 13 49 33](https://user-images.githubusercontent.com/17420811/144552640-c733d7af-7a09-4dd7-a6cf-25f0a4474b34.png)

---

📜  Probably solution

Fix by WC-admin will need to continue waiting for the next L-2 to include the patch. To move forward with this PR, I tried changing `FREE_LISTINGS_PROGRAM_ID` constant to `-1` and it seems to work correctly, but I didn't check if there are new side effects afterward.

tomalec commented:
Thanks for finding that!

This looks almost like a regression for https://github.com/woocommerce/woocommerce-admin/pull/7028

I think it really unfortunate, we cannot use either the old or new version of `@woocommerce/components.FilterPicker`.
Maybe we need to PR the full tests coverage for the component to wc-admin repo, before we start using it here :|

eason9487 commented:
@tomalec, thanks for the work! Since this PR has been inactive for a long time and still has incompatibility concerns, it is unlikely to be resolved in the near future, whether through upstream fixes or workarounds in this plugin. Therefore, I'm closing it for now.
